packages/plugins/i18n/server/services/localizations.js

```diff
@@ -11,7 +11,14 @@ const { getService } = require('../utils');
 const assignDefaultLocale = async (data) => {
   const { getDefaultLocale } = getService('locales');
 
-  if (isNil(data.locale)) {
+  if (isArray(data) && data.some((entry) => !entry.locale)) {
+    const defaultLocale = await getDefaultLocale();
+    data.forEach((entry) => {
+      if (isNil(entry.locale)) {
+        entry.locale = defaultLocale;
+      }
+    });
+  } else if (isNil(data.locale)) {
     data.locale = await getDefaultLocale();
   }
 };
```

Comment:
It's not related to this specific PR, but maybe we could call `assignDefaultLocale` to `assignDefaultLocaleToEntries`, it seems a bit unclear where are we assigning the default locale. wdyt?

Reply:
Yes I'm not against :)